FATAL EXCEPTION: main - Attempt to invoke interface method 'com.getcapacitor.JSObject io.capawesome.capacitorjs.plugins.firebase.firestore.interfaces.Result.toJSObject()' on a null object reference

### Discussed in https://github.com/capawesome-team/capacitor-firebase/discussions/468

<div type='discussions-op-text'>

<sup>Originally posted by **FrancescoIoppolo** October 18, 2023</sup>
On Android I get this error:
Can you help me?

**FATAL EXCEPTION: main
                                                                                                    Process: digipetx.app, PID: 16119
                                                                                                    java.lang.NullPointerException: Attempt to invoke interface method 'com.getcapacitor.JSObject io.capawesome.capacitorjs.plugins.firebase.firestore.interfaces.Result.toJSObject()' on a null object reference
                                                                                                    	at io.capawesome.capacitorjs.plugins.firebase.firestore.FirebaseFirestorePlugin$3.success(FirebaseFirestorePlugin.java:129)
                                                                                                    	at io.capawesome.capacitorjs.plugins.firebase.firestore.FirebaseFirestore.lambda$getDocument$4(FirebaseFirestore.java:84)
                                                                                                    	at io.capawesome.capacitorjs.plugins.firebase.firestore.FirebaseFirestore$$ExternalSyntheticLambda1.onSuccess(Unknown Source:4)
                                                                                                    	at com.google.android.gms.tasks.zzm.run(com.google.android.gms:play-services-tasks@@18.0.2:1)
                                                                                                    	at android.os.Handler.handleCallback(Handler.java:938)
                                                                                                    	at android.os.Handler.dispatchMessage(Handler.java:99)
                                                                                                    	at android.os.Looper.loop(Looper.java:264)
                                                                                                    	at android.app.ActivityThread.main(ActivityThread.java:8312)
                                                                                                    	at java.lang.reflect.Method.invoke(Native Method)
                                                                                                    	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:632)
                                                                                                    	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1049)**</div>

yes, solve it! great

now I get this: 

**FATAL EXCEPTION: CapacitorPlugins
                                                                                                     Process: digipetx.app, PID: 28035
                                                                                                     java.lang.RuntimeException: java.lang.reflect.InvocationTargetException
                                                                                                     at com.getcapacitor.Bridge.lambda$callPluginMethod$0(Bridge.java:783)
                                                                                                     at com.getcapacitor.Bridge.$r8$lambda$ehFTi5f4HhVNFKTbCKAYDkpQYRA(Unknown Source:0)
                                                                                                     at com.getcapacitor.Bridge$$ExternalSyntheticLambda3.run(Unknown Source:8)
                                                                                                     at android.os.Handler.handleCallback(Handler.java:938)
                                                                                                     at android.os.Handler.dispatchMessage(Handler.java:99)
                                                                                                     at android.os.Looper.loop(Looper.java:264)
                                                                                                     at android.os.HandlerThread.run(HandlerThread.java:67)
                                                                                                     Caused by: java.lang.reflect.InvocationTargetException
                                                                                                     at java.lang.reflect.Method.invoke(Native Method)
                                                                                                     at com.getcapacitor.PluginHandle.invoke(PluginHandle.java:138)
                                                                                                     at com.getcapacitor.Bridge.lambda$callPluginMethod$0(Bridge.java:774)
                                                                                                     at com.getcapacitor.Bridge.$r8$lambda$ehFTi5f4HhVNFKTbCKAYDkpQYRA(Unknown Source:0)
                                                                                                     at com.getcapacitor.Bridge$$ExternalSyntheticLambda3.run(Unknown Source:8)
                                                                                                     at android.os.Handler.handleCallback(Handler.java:938)
                                                                                                     at android.os.Handler.dispatchMessage(Handler.java:99)
                                                                                                     at android.os.Looper.loop(Looper.java:264)
                                                                                                     at android.os.HandlerThread.run(HandlerThread.java:67)
                                                                                                     Caused by: java.lang.NoClassDefFoundError: Failed resolution of: Landroid/support/v4/app/FragmentActivity;
                                                                                                     at com.google.android.gms.auth.api.signin.internal.zzi.zzd(Unknown Source:0)
                                                                                                     at com.google.android.gms.auth.api.signin.GoogleSignInClient.getSignInIntent(Unknown Source:34)
                                                                                                     at io.capawesome.capacitorjs.plugins.firebase.authentication.handlers.GoogleAuthProviderHandler.signIn(GoogleAuthProviderHandler.java:40)
                                                                                                     at io.capawesome.capacitorjs.plugins.firebase.authentication.FirebaseAuthentication.signInWithGoogle(FirebaseAuthentication.java:456)
                                                                                                     at io.capawesome.capacitorjs.plugins.firebase.authentication.FirebaseAuthenticationPlugin.signInWithGoogle(FirebaseAuthenticationPlugin.java:583)
                                                                                                     at java.lang.reflect.Method.invoke(Native Method)
                                                                                                     at com.getcapacitor.PluginHandle.invoke(PluginHandle.java:138)
                                                                                                     at com.getcapacitor.Bridge.lambda$callPluginMethod$0(Bridge.java:774)**

--- a/.changeset/tame-parents-burn.md
+++ b/.changeset/tame-parents-burn.md
@@ -1,0 +1,5 @@
+---
+'@capacitor-firebase/firestore': patch
+---
+
+fix(android): `NullPointerException` in `getDocument(...)`

--- a/packages/firestore/android/src/main/java/io/capawesome/capacitorjs/plugins/firebase/firestore/FirebaseFirestore.java
+++ b/packages/firestore/android/src/main/java/io/capawesome/capacitorjs/plugins/firebase/firestore/FirebaseFirestore.java
@@ -77,12 +77,8 @@ public class FirebaseFirestore {
             .get()
             .addOnSuccessListener(
                 documentSnapshot -> {
-                    if (documentSnapshot.exists()) {
-                        GetDocumentResult result = new GetDocumentResult(documentSnapshot);
-                        callback.success(result);
-                    } else {
-                        callback.success(null);
-                    }
+                    GetDocumentResult result = new GetDocumentResult(documentSnapshot);
+                    callback.success(result);
                 }
             )
             .addOnFailureListener(exception -> callback.error(exception));

--- a/packages/firestore/android/src/main/java/io/capawesome/capacitorjs/plugins/firebase/firestore/classes/results/GetDocumentResult.java
+++ b/packages/firestore/android/src/main/java/io/capawesome/capacitorjs/plugins/firebase/firestore/classes/results/GetDocumentResult.java
@@ -4,6 +4,7 @@ import com.getcapacitor.JSObject;
 import com.google.firebase.firestore.DocumentSnapshot;
 import io.capawesome.capacitorjs.plugins.firebase.firestore.FirebaseFirestoreHelper;
 import io.capawesome.capacitorjs.plugins.firebase.firestore.interfaces.Result;
+import org.json.JSONObject;
 
 public class GetDocumentResult implements Result {
 
@@ -14,7 +15,12 @@ public class GetDocumentResult implements Result {
     }
 
     public JSObject toJSObject() {
-        JSObject snapshotDataResult = FirebaseFirestoreHelper.createJSObjectFromMap(documentSnapshot.getData());
+        Object snapshotDataResult;
+        if (documentSnapshot.exists()) {
+            snapshotDataResult = FirebaseFirestoreHelper.createJSObjectFromMap(documentSnapshot.getData());
+        } else {
+            snapshotDataResult = JSONObject.NULL;
+        }
 
         JSObject snapshotResult = new JSObject();
         snapshotResult.put("id", documentSnapshot.getId());

